test/unit/commandLineParser.spec.ts

```diff
@@ -1,235 +1,177 @@
+            project: "tsconfig.json",
+            noHeader: true,
+            target: ts.ScriptTarget.ES3,
+            luaTarget: tstl.LuaTarget.Lua53,
+        });
+    });
 
-test.each([
-    { args: [""], expected: LuaTarget.LuaJIT },
-    { args: ["--luaTarget", "5.1"], expected: LuaTarget.Lua51 },
-    { args: ["--luaTarget", "5.2"], expected: LuaTarget.Lua52 },
-    { args: ["--luaTarget", "jit"], expected: LuaTarget.LuaJIT },
-    { args: ["--luaTarget", "JIT"], expected: LuaTarget.LuaJIT },
-    { args: ["--luaTarget", "5.3"], expected: LuaTarget.Lua53 },
-])("CLI parser luaTarget (%p)", ({ args, expected }) => {
-    const result = parseCommandLine(args);
-    if (result.isValid === true) {
-        expect(result.result.options.luaTarget).toBe(expected);
-    } else {
-        expect(result.isValid).toBeTruthy();
-    }
-});
+    test("should error on unknown options", () => {
+        const result = tstl.parseCommandLine(["--unknownOption"]);
 
-test.each([
-    { args: ["-lt", "5.1"], expected: LuaTarget.Lua51 },
-    { args: ["-lt", "5.2"], expected: LuaTarget.Lua52 },
-    { args: ["-lt", "jit"], expected: LuaTarget.LuaJIT },
-    { args: ["-lt", "JIT"], expected: LuaTarget.LuaJIT },
-    { args: ["-lt", "5.3"], expected: LuaTarget.Lua53 },
-])("CLI parser luaTarget (%p)", ({ args, expected }) => {
-    const result = parseCommandLine(args);
-    if (result.isValid === true) {
-        expect(result.result.options.luaTarget).toBe(expected);
-    } else {
-        expect(result.isValid).toBeTruthy();
-    }
-});
+        expect(result.errors).toHaveDiagnostics();
+    });
 
-test("CLI parser invalid luaTarget", () => {
-    const result = parseCommandLine(["--luatTarget", "invalid"]);
-    expect(result.isValid).toBe(false);
-});
+    test("should parse options case-insensitively", () => {
+        const result = tstl.parseCommandLine(["--NoHeader"]);
 
-test.each([
-    { args: [""], expected: false },
-    { args: ["--noHeader", "true"], expected: true },
-    { args: ["--noHeader", "false"], expected: false },
-    { args: ["--noHeader"], expected: true },
-    { args: ["--noHeader", "--noHoisting"], expected: true },
-])("CLI parser noHeader (%p)", ({ args, expected }) => {
-    const result = parseCommandLine(args);
-    if (result.isValid === true) {
-        expect(result.result.options.noHeader).toBe(expected);
-    } else {
-        expect(result.isValid).toBeTruthy();
-    }
-});
+        expect(result.errors).not.toHaveDiagnostics();
+        expect(result.options.noHeader).toBe(true);
+    });
+
+    describe("enum options", () => {
+        test("should parse enums", () => {
+            const result = tstl.parseCommandLine(["--luaTarget", "5.1"]);
+
+            expect(result.errors).not.toHaveDiagnostics();
+            expect(result.options.luaTarget).toBe(tstl.LuaTarget.Lua51);
+        });
+
+        test("should be case-insensitive", () => {
+            for (const value of ["jit", "JiT", "JIT"]) {
```

Comment:
Why not test.each?

Reply:
My thought is that if `test.each` would break that test to 3 separate tests, a single test won't be able to prove a common `"should be case-insensitive"` contract

Reply:
That is a good point, I'm fine with keeping it like this.